server/index.js

```diff
@@ -46,14 +46,131 @@ app.get('/api/products/:productId', (req, res, next) => {
     db.query(getAllProductsSql, value)
       .then(result => {
         if (!result.rows[0]) {
-          next(new ClientError(`cannot ${req.method} find id ${productId}`), 404);
+          next(new ClientError(`cannot ${req.method}/find id ${productId}`), 404);
         } else {
           return res.json(result.rows[0]);
         }
       })
       .catch(err => next(err));
   }
 });
+// GET endpoint for /api/cart
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session.cartId) {
+    return res.status(200).json([]);
+  }
+
+  const getAllProductsFromCartSql = `
+      SELECT "c"."cartItemId",
+             "c"."price",
+             "p"."productId",
+             "p"."image",
+             "p"."name",
+             "p"."shortDescription"
+        FROM "cartItems" as "c"
+        JOIN "products" as "p" using ("productId")
+       WHERE "c"."cartId" = $1
+  `;
+  const value = [req.session.cartId];
+
+  db.query(getAllProductsFromCartSql, value)
+    .then(result => {
+      res.status(200).json(result.rows);
+    })
+    .catch(err => next(err));
+});
+
+// POST endpoint for /api/cart
+app.post('/api/cart/', (req, res, next) => {
+  const { productId } = req.body;
+
+  if (productId < 0) {
+    res.status(400).json({
+      error: 'id must be a positive integer'
+    });
+  }
+
+  const getProductPriceSql = `
+      SELECT "price"
+        FROM "products"
+       WHERE "productId" = $1
+    `;
+  const value = [productId];
+
+  db.query(getProductPriceSql, value)
+  // 1. .then()
+    .then(result => {
+      if (result.rows.length < 0) {
+        next(new ClientError(`Cannot find a product with the productId=${productId}`), 400);
```

Comment:
Because we're in a Promise chain, you should `throw` the error, instead of passing it to `next()` . `next` does not end the Promise chain!